app.go

```diff
@@ -286,10 +289,16 @@ type App struct {
 	// Decides how we react to errors when building the graph.
 	errorHooks []ErrorHandler
 	validate   bool
+
 	// Used to signal shutdowns.
-	donesMu     sync.Mutex // guards dones and shutdownSig
-	dones       []chan os.Signal
-	shutdownSig os.Signal
+	shutdownMu   sync.Mutex
+	shutdownSig  *ShutdownSignal
+	sigReceivers []signalReceiver
+	signalOnce   sync.Once
+
+	// Used to make sure Start/Stop is called only once.
```

Comment:
do you need this to be included in this PR? I'm asking because this stuff was supposed to ship with the previous version (v1.17.2) and it broke some users: see https://github.com/uber-go/fx/issues/945 and https://github.com/uber-go/fx/pull/950 for context.